ipi_tests/regression_tests/runstools.py

```diff
@@ -0,0 +1,114 @@
+from tempfile import TemporaryDirectory
+import tempfile
+
+
+def get_info_test(parent):
+    folders = [x[0] for x in os.walk(parent)]
+    reg_tests = list()
+
+    for ff in folders:
+        if os.path.isfile(Path(ff) / "input.xml"):
+            driver_info = list()
+            try:
+                with open(Path(ff) / "driver.txt") as f:
+                    ncount = 0
+                    while True:
+                        line = f.readline()
+                        if not line:
+                            break
+                        else:
+                            driver, address, port, mode = line.split()
+                            driver_info.append((driver, address, port, mode))
+                            ncount += 1
+                    reg_tests.append((ff, driver_info))
+                    if ncount == 0:
+                        raise ValueError("driver.txt is empty")
+            except FileNotFoundError:
+                raise FileNotFoundError(
+                    "({}) An input.xml file was found but a driver.txt not".format(ff)
+                )
+            except ValueError:
+                raise ValueError(
+                    'Please specify "model address      port mode" inside driver.txt'
+                )
+
+    return reg_tests
+
+
+class Runner(object):
+    def __init__(self, parent, check_errors=True, check_prop=True):
+        self.parent = parent
+        self.check_error = check_errors
+        self.check_prop = check_prop
+
+    def _run(self, cmd1, cmd2, cwd):
+
+        try:
+            for filename in glob.glob("/tmp/ipi_*"):
+                os.remove(filename)
+        except:
+            pass
+
+        try:
+            self.tmp_dir = Path(tempfile.mkdtemp())
+            files = os.listdir(self.parent / cwd)
+            for f in files:
+                shutil.copy(self.parent / cwd / f, self.tmp_dir)
+
+            ipi = sp.Popen(
+                cmd1, cwd=(self.tmp_dir), shell=True, stdout=sp.PIPE, stderr=sp.PIPE
+            )
+            time.sleep(2)
+            driver = list()
+            for cmd in cmd2:
+                driver.append(sp.Popen(cmd, shell=True, stdout=sp.PIPE, stderr=sp.PIPE))
+
+            if self.check_error:
+                self._check_error(ipi)
+            if self.check_prop:
+                self._check_properties(cwd)
+
+        except sp.TimeoutExpired:
+            raise RuntimeError(
+                "Time is out. Aborted during {} test. \
+              Error {}".format(
+                    str(cwd), ipi.communicate(timeout=2)[0]
+                )
+            )
+
+        except FileNotFoundError:
+            raise FileNotFoundError("{}".format(str(cwd)))
+
+        except ValueError:
+            raise ValueError("{}".format(str(cwd)))
+
+    def _check_error(self, ipi):
+        ipi_error = ipi.communicate(timeout=30)[1].decode("ascii")
+        print(ipi_error)
+        assert "" == ipi_error
+
+    def _check_properties(self, cwd):
```

Comment:
What about other properties ? Currently it looks like it only checks for a similaton.out file. What if there are more than one property files ?

Reply:
Here I agree, but I think other properties can be added later on. This is supposed to be a minimal "skeleton" of the infrastructure, so definitely adding a doc string with a hint as to how one could enhance this function (other properties) would be a good way to go.

Reply:
I added some dummy functions. I have some ideas but it would be better to discuss them first.